src/pages/Staking/LockTimeField/index.tsx

```diff
@@ -0,0 +1,158 @@
+
+import * as React from 'react';
+import clsx from 'clsx';
+
+import {
+  TextFieldHelperText,
+  TextFieldLabel
+} from 'components/TextField';
+import NumberInput, { Props as NumberInputProps } from 'components/NumberInput';
+import { STAKE_LOCK_TIME } from 'config/relay-chains';
+import {
+  POLKADOT,
+  KUSAMA
+} from 'utils/constants/relay-chain-names';
+
+// MEMO: inspired by https://medium.com/codex/making-html-5-numeric-inputs-only-accept-integers-d3d117973d56
+const intRx = /\d/;
+const handleLockTimeChange = (event: KeyboardEvent) => {
+  if (
+    (event.key.length > 1) ||
+    ((event.key === '-') && (!(event.currentTarget as HTMLInputElement)?.value?.length)) ||
+    intRx.test(event.key)
+  ) return;
+
+  event.preventDefault();
+};
+
+const LABEL_TEXT_COLOR_CLASSES = clsx(
+  // TODO: placeholder color
+  { 'text-interlayTextSecondaryInLightMode': process.env.REACT_APP_RELAY_CHAIN_NAME === POLKADOT },
+  // TODO: placeholder color
+  { 'dark:text-kintsugiTextSecondaryInDarkMode': process.env.REACT_APP_RELAY_CHAIN_NAME === KUSAMA }
+);
+
+interface CustomProps {
+  optional?: boolean;
+  error?: boolean;
+  helperText?: JSX.Element | string;
+}
+
+type Ref = HTMLInputElement;
+const LockTimeField = React.forwardRef<Ref, CustomProps & NumberInputProps>(({
+  optional,
+  id,
+  name,
+  error,
+  helperText,
+  ...rest
+}, ref): JSX.Element => {
+  const wrappingRef = React.useRef<HTMLInputElement | null>(null);
+
+  React.useEffect(() => {
+    if (!wrappingRef) return;
+    if (!wrappingRef.current) return;
+
+    const wrappingRefCurrent = wrappingRef.current;
+
+    wrappingRefCurrent.addEventListener('keydown', handleLockTimeChange);
+
+    return () => {
+      wrappingRefCurrent.removeEventListener('keydown', handleLockTimeChange);
+    };
+  }, []);
+
+  return (
+    <div>
+      <TextFieldLabel
+        htmlFor={id}
+        className={clsx(
+          '!text-xs',
+          LABEL_TEXT_COLOR_CLASSES
+        )}
+        required={optional === false}>
```

Comment:
Yes, actually, it's set like so on purpose. `optional` is `boolean` | `undefined`.
`undefined` means that we are not sure yet if it's really optional or not because we are loading some data which is to be used for determining whether it's optional.
If we set `!optional`, while the data being loaded from the beginning, there appears an asterisk like `Max 96 Weeks *`. And once it's determined to be optional (`true`), then it will disappear so it would cause flashing which could impact the UX.